xbmc/guilib/GUIIncludes.h

```diff
@@ -47,13 +47,13 @@ class CGUIIncludes
   void Clear();
 
   /*!
-   \brief Load all include components (defaults, constants, variables, expressions and includes)
-    from the given \code{file}.
-   
-   \param file the file to load
-   \return true if the file was loaded otherwise false
-   */
-  bool Load(const std::string &file);
+  \brief Load all include components (defaults, constants, variables, expressions and includes)
+  from the main entrypoint \code{file}. Flattens nested expressions and expressions in variable
+  conditions after loading all other included files.
+
+  \param file the file to load
+  */
+  void LoadEntrypoint(const std::string &file);
```

Comment:
`Load` is called for every single included file, and flattening all previously loaded expressions/variables when loading each new file is just wasting cycles when they can all be flattened once after everything is loaded. That could add an unnecessary delay when loading and reloading a skin, depending on hardware and how skinners lay out their include files. The current code works no matter how many include files or in what order the skinner chooses to include them, simply and efficiently.

Imagine a skin with variables defined in _variables.xml_ and expressions in _expressions.xml_. If _variables.xml_ is included first then flatten when loading it will have to give  `CGUIInfoLabel::ReplaceSpecialKeywordReferences` the same `$EXP[whatever]` it is trying to replace (so it can try again after the next file is loaded), accomplishing nothing but wasting cycles. The same thing will happen when loading every included file between here and _expressions.xml_, compounding the issue.

I could move the flatten calls behind a bool parameter `flatten` on the `Load` method that would only be true for the main _include.xml_ file (and "file" includes from `ResolveIncludes`, which I've missed currently), if that is better than this new method.